purchases/src/main/kotlin/com/revenuecat/purchases/common/Backend.kt

```diff
@@ -441,14 +453,73 @@ internal class Backend(
         synchronized(this@Backend) {
             diagnosticsCallbacks.addCallback(
                 call,
-                diagnosticsDispatcher,
+                eventsDispatcher,
                 cacheKey,
                 onSuccessHandler to onErrorHandler,
                 Delay.LONG,
             )
         }
     }
 
+    @RequiresApi(Build.VERSION_CODES.N)
+    fun postPaywallEvents(
+        paywallEventRequest: PaywallEventRequest,
+        onSuccessHandler: () -> Unit,
+        onErrorHandler: (error: PurchasesError, shouldMarkAsSynced: Boolean) -> Unit,
+    ) {
+        val body = PaywallEventsManager.json.encodeToJsonElement(paywallEventRequest).asMap()
+        if (body == null) {
+            onErrorHandler(
+                PurchasesError(
+                    PurchasesErrorCode.UnknownError,
+                    "Error encoding paywall event request",
+                ).also { errorLog(it) },
+                true,
+            )
+            return
+        }
+        val call = object : Dispatcher.AsyncCall() {
+            override fun call(): HTTPResult {
+                return httpClient.performRequest(
+                    appConfig.paywallEventsURL,
+                    Endpoint.PostPaywallEvents,
+                    body,
+                    postFieldsToSign = null,
+                    backendHelper.authenticationHeaders,
+                )
+            }
+
+            override fun onError(error: PurchasesError) {
```

Comment:
Also outside the scope of this PR, but there is a lot of repetition on how we make requests as well that would be useful to abstract.

Reply:
Absolutely agree.